mesonbuild/backend/vs2010backend.py

```diff
@@ -1071,7 +1071,8 @@ def gen_vcxproj(self, target, ofname, guid):
         # Size-preserving flags
         if '/Os' in o_flags:
             ET.SubElement(clconf, 'FavorSizeOrSpeed').text = 'Size'
-        else:
+        # Note: setting FavorSizeOrSpeed with clang-cl overrides /Od and makes debugging difficult.
+        elif not '/Od' in o_flags:
```

Comment:
This will potentially change in #8179 as we won't set `/Od` by default any more. How will it affect this change?

Also, the documentation says that there are three choices: https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.vcprojectengine.favorsizeorspeedoption?view=visualstudiosdk-2019

MSDN docs do not seem to say if you should have the value `none` in that case or is a missing value interpreted as `none`.

Reply:
Doesn't seem that #8179 affects it. For `--buildtype=debug`:

| | 0.56.2 | marstaik:no_ZI | lukester1975:fix-clang-cl-vs-backend |
|-|-|-|-|
FavorSizeOrSpeed | Speed | Speed | not present |
BasicRuntimeChecks | EnableFastChecks | not present | EnableFastChecks |
Optimization | Disabled | not present | Disabled |

For the second point: a default "Hello World" vcxproj from VS doesn't have `FavorSizeOrSpeed` at all (i.e. debug or release). So missing is the same as `none`, I guess (though the IDE's wording is better - `neither`!).

Reply:
FYI, using ninja there is no `/Ot` for a debug build:

```
"/MDd" "/nologo" "/showIncludes" "/W2" "/EHsc" "/std:c++14" "/permissive-" "/Zi" "/Ob0" 
"/Od" "/RTC1" "/Od" "/Ob0" "/Zi" "/arch:AVX"
```

VS backend:
```
/GS /W2 /Gy /Zi /Od /fp:precise /D "_MBCS" /WX- /Gd /MDd /EHsc /nologo /Ot
/diagnost "/EHsc" "/std:c++14" "/permissive-" "/arch:AVX" 
```

(ignoring include paths, pdbs and other intermediates).

For this particular example, the differences boil down to:

VS: `/GS /Gy /fp:precise /D "_MBCS" /WX- /Gd /Ot`

Ninja: `"/Ob0" "/RTC1" "/Zi"`